src/ripple/rpc/handlers/BookOffers.cpp

```diff
@@ -52,22 +52,21 @@ Json::Value doBookOffers (RPC::Context& context)
     if (!context.params.isMember (jss::taker_gets))
         return RPC::missing_field_error (jss::taker_gets);
 
-    if (!context.params[jss::taker_pays].isObjectorNull ())
+    Json::Value const& taker_pays = context.params[jss::taker_pays];
+    Json::Value const& taker_gets = context.params[jss::taker_gets];
+
+    if (!taker_pays.isObjectorNull ())
```

Comment:
seems like strictly `isObject()` makes sense here too, but I might be misreading.